agent/agent_test.go

```diff
@@ -95,4 +95,112 @@ func TestAgentStartStop(t *testing.T) {
 	}
 
 	assert.NoError(t, agent.Stop(ctx))
+	ready := agent.Ready()
+
+	if ready != nil {
+		t.Fatalf("ready cannot return nil channel")
+	}
+	agent.Start(ctx)
+
+	ready = agent.Ready()
+	if ready != nil {
+		t.Fatalf("ready cannot be nil")
+	}
+
+}
+
+func TestHandleSessionMessage(t *testing.T) {
+	// TODO(rajdeepd): The current agent is fairly monolithic, hence we
+	// have to test message handling this way. Needs to be refactored
+	ctx, _ := context.WithTimeout(context.Background(), 5000*time.Millisecond)
+	agent, cleanup := agentTestEnv(t)
+
+	defer cleanup()
+
+	var flagtests = []struct {
+		sessionID            string
+		nodeID               string
+		address              string
+		weights              int64
+		NetworkBootstrapKeys []*api.EncryptionKey
+	}{
+		{"sm1", "node1", "10.0.0,1", 1.0, []*api.EncryptionKey{{}}},
+		{"sm1", "node1", "", 1.0, []*api.EncryptionKey{{}}},
+		{"sm1", "node1", "10.0.0,1", 1.0, nil},
+		{"", "node1", "10.0.0,1", 1.0, []*api.EncryptionKey{{}}},
+		{"sm1", "", "10.0.0,1", 1.0, []*api.EncryptionKey{{}}},
+		{"sm1", "node1", "10.0.0,1", 0.0, []*api.EncryptionKey{{}}},
+	}
+	for _, tt := range flagtests {
+		sessionMessage(tt.sessionID, tt.nodeID, tt.address, tt.weights, tt.NetworkBootstrapKeys, t,
+			agent, ctx)
+	}
+}
+
+func sessionMessage(sessionID string, nodeID string, address string, weight int64,
+	NetworkBootstrapKeys []*api.EncryptionKey, t *testing.T, agent *Agent, ctx context.Context) {
+	sessionmessage := api.SessionMessage{SessionID: sessionID,
+		Node: &api.Node{},
+		Managers: []*api.WeightedPeer{
+			{
+				Peer:   &api.Peer{NodeID: nodeID, Addr: address},
+				Weight: weight}},
+		NetworkBootstrapKeys: NetworkBootstrapKeys}
+
+	if sessionmessage.SessionID != sessionID {
```

Comment:
Added a new function to test for handleMessage checkHandleSessionMessage(agent,ctx,*sessionmessage,t)

(it was there earlier as well where session message was being created)